### ISSUE TITLE
NULL pointer dereference in stb_image.h

### ISSUE THREAD

**Issue by eldstal** [saitoha/libsixel#160]
## Vulnerable versions
* saitoha/libsixel at the latest (6a5be8b72d84037b83a5ea838e17bcf372ab1d5f) commit
* libsixel/libsixel at the latest (bc93c8ce570b35b8f3bb429ea5892ac260e4c8b1) commit

## Steps to reproduce
`img2sixel stbio_1561_poc.bin`

Input file (a malformed PICT-format image) is [attached](https://github.com/saitoha/libsixel/files/7715038/stbi_1561_poc.zip).

## Cause
Segmentation fault in `stbi__convert_format` at `stb_image.h:1561`:

```c
   switch (STBI__COMBO(img_n, req_comp)) {
     /* ... */
     STBI__CASE(4,3) { dest[0]=src[0],dest[1]=src[1],dest[2]=src[2]; } break;
     /* ... */
   }
```

The `src` pointer is `NULL`, as passed in from `stbi__pic_load`.

The source of the NULL pointer is the malloc at line `6120`:
```c
   result = (stbi_uc *) stbi__malloc_mad4(x, y, 4, 0);
```
whose output is never checked for `NULL`. The x and y dimensions (39168, 5888) are read
directly from the input file, and they pass the check in `stbi__mad3sizes_valid` which
only checks for integer overflow.

The total size of the allocated buffer is `39168 * 5888 * 4` and allocation fails.

## Impact
Denial of service is the only obvious impact.

## Mitigation
`stb_image` starting at version 2.27 (50072f66589f52f51eb5b3f56b9272ea8ec1fdac) include a check for this condition. libsixel should be brought up-to-date with this version if possible.

If not, backport [the check](https://github.com/nothings/stb/blob/af1a5bc352164740c1cc1354942b1c6b72eacb8a/stb_image.h#L6448) as well as similar error checks for other malloc calls.

**Comment by eldstal** [comment]
This same report was also [submitted in the fork](https://github.com/libsixel/libsixel/issues/51).

**Comment by eldstal** [comment]
Fixed by @dankamongmen in commit f283ece716e925c4a4a8f5b94b99534b0b11d9be of the libsixel/libsixel port.

**Comment by eldstal** [comment]
This vulnerability has been assigned [CVE-2021-45340](https://www.cve.org/CVERecord?id=CVE-2021-45340).
